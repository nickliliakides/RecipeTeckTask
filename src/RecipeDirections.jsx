import React from 'react';
import Loader from './Loader';

const RecipeDirections = ({ recipe, loading }) => {
  return recipe && !loading ? (
    <div className='recipe'>
      <figure className='recipe__fig'>
        <img src={recipe.source_url} alt='Tomato' className='recipe__img' />
        <h1 className='recipe__title'>
          <span>{recipe.title}</span>
        </h1>
      </figure>
      <div className='recipe__details'>
        <div className='recipe__info'>
          <svg className='recipe__info-icon'>
            <use href='img/icons.svg#icon-stopwatch'></use>
          </svg>
          <span className='recipe__info-data recipe__info-data--minutes'>
            {`${recipe.prep_time_min} + ${recipe.cook_time_min}`}
          </span>
          <span className='recipe__info-text'> minutes (prep + cook)</span>
        </div>
      </div>

      <div className='recipe__directions'>
        <h2 className='heading-2'>How to cook it</h2>
        {recipe.directions.map((dir, i) => (
          <p key={i} className='recipe__directions-text'>
            {dir}
          </p>
        ))}
        <hr />
        <p className='recipe__directions-last-text'>
          This recipe was carefully designed and tested by
          <span className='recipe__by'>&nbsp;{recipe.author.name}</span>. Please
          check out more recipes at their website.
        </p>
        <a
          className='btn-small recipe__btn'
          href={recipe.author.url}
          target='_blank'
          rel='noreferrer'
        >
          <span>Directions</span>
          <svg className='search__icon'>
            <use href='img/icons.svg#icon-triangle-right'></use>
          </svg>
        </a>
      </div>
    </div>
  ) : (
    <Loader />
  );
};

export default RecipeDirections;
