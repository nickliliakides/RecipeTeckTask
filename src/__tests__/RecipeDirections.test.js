import React from 'react';
import { shallow } from 'enzyme';
import RecipeDirections from '../RecipeDirections';
import recipes from '../mockRecipes';

let wrapper;

const defaultProps = {
  recipe: recipes[0],
  loading: false,
};

describe('<RecipeDirections /> component', () => {
  beforeEach(() => {
    wrapper = shallow(<RecipeDirections {...defaultProps} />);
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('should matches its snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
