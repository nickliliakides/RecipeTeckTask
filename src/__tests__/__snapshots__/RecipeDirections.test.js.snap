// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeDirections /> component should matches its snapshot 1`] = `
<div
  className="recipe"
>
  <figure
    className="recipe__fig"
  >
    <img
      alt="Tomato"
      className="recipe__img"
      src="https://imagesvc.meredithcorp.io/v3/mm/image?url=https%3A%2F%2Fimages.media-allrecipes.com%2Fuserphotos%2F8224154.jpg&q=85"
    />
    <h1
      className="recipe__title"
    >
      <span>
        Vegetarian Korma
      </span>
    </h1>
  </figure>
  <div
    className="recipe__details"
  >
    <div
      className="recipe__info"
    >
      <svg
        className="recipe__info-icon"
      >
        <use
          href="img/icons.svg#icon-stopwatch"
        />
      </svg>
      <span
        className="recipe__info-data recipe__info-data--minutes"
      >
        25 + 30
      </span>
      <span
        className="recipe__info-text"
      >
         minutes (prep + cook)
      </span>
    </div>
  </div>
  <div
    className="recipe__directions"
  >
    <h2
      className="heading-2"
    >
      How to cook it
    </h2>
    <p
      className="recipe__directions-text"
      key="0"
    >
      Heat the oil in a skillet over medium heat. Stir in the onion, and cook until tender.
    </p>
    <p
      className="recipe__directions-text"
      key="1"
    >
      Mix in ginger and garlic, and continue cooking 1 minute.
    </p>
    <p
      className="recipe__directions-text"
      key="2"
    >
      Mix potatoes, carrots, jalapeno, cashews, and tomato sauce. Season with salt and curry powder. Cook and stir 10 minutes, or until potatoes are tender.
    </p>
    <p
      className="recipe__directions-text"
      key="3"
    >
      Stir peas, green bell pepper, red bell pepper, and cream into the skillet. Reduce heat to low, cover, and simmer 10 minutes.
    </p>
    <p
      className="recipe__directions-text"
      key="4"
    >
      Garnish with cilantro to serve.
    </p>
    <hr />
    <p
      className="recipe__directions-last-text"
    >
      This recipe was carefully designed and tested by
      <span
        className="recipe__by"
      >
        Â 
        Yakuta
      </span>
      . Please check out more recipes at their website.
    </p>
    <a
      className="btn-small recipe__btn"
      href="http://allrecipes.com/cook/116722/profile.aspx"
      rel="noreferrer"
      target="_blank"
    >
      <span>
        Directions
      </span>
      <svg
        className="search__icon"
      >
        <use
          href="img/icons.svg#icon-triangle-right"
        />
      </svg>
    </a>
  </div>
</div>
`;
